numpyro/contrib/module.py

```diff
@@ -71,6 +81,166 @@ def haiku_module(name, nn, input_shape=None):
+
+        has parameters `{'inner': {'dense': {'kernel': param_value}}}`. In the argument
+        `prior`, to specify `kernel` parameter, we join the path to it using dots:
+        `prior={"inner.dense.kernel": param_prior}`.
+
+    :param str name: name of NumPyro module
+    :param flax.nn.Module: the module to be registered with NumPyro
+    :param prior: a NumPyro distribution or a Python dict with parameter names as keys and
+        respective distributions as values. For example::
+
+            net = random_flax_module("net",
+                                     flax.nn.Dense.partial(features=1),
+                                     prior={"bias": dist.Cauchy(), "kernel": dist.Normal()},
+                                     input_shape=(4,))
+
+    :type param: dict or ~numpyro.distributions.Distribution
+    :param tuple input_shape: shape of the input taken by the neural network.
+    :returns: a sampled module
+
+    **Example**
+
+    .. doctest::
+
+        # NB: this example is ported from https://github.com/ctallec/pyvarinf/blob/master/main_regression.ipynb
+        >>> import numpy as np; np.random.seed(0)
+        >>> import tqdm
+        >>> from flax import nn
+        >>> from jax import jit, random
+        >>> import numpyro
+        >>> import numpyro.distributions as dist
+        >>> from numpyro.contrib.module import random_flax_module
+        >>> from numpyro.infer import ELBO, Predictive, SVI, autoguide, init_to_feasible
+        >>>
+        >>> class Net(nn.Module):
+        ...     def apply(self, x, n_units):
+        ...         x = nn.Dense(x[..., None], features=n_units)
+        ...         x = nn.relu(x)
+        ...         x = nn.Dense(x, features=n_units)
+        ...         x = nn.relu(x)
+        ...         mean = nn.Dense(x, features=1)
+        ...         rho = nn.Dense(x, features=1)
+        ...         return mean.squeeze(), rho.squeeze()
+        >>>
+        >>> def generate_data(n_samples):
+        ...     x = np.random.normal(size=n_samples)
+        ...     y = np.cos(x * 3) + np.random.normal(size=n_samples) * np.abs(x) / 2
+        ...     return x, y
+        >>>
+        >>> def model(x, y=None, batch_size=None):
+        ...     module = Net.partial(n_units=32)
+        ...     net = random_flax_module("nn", module, dist.Normal(0, 0.1), input_shape=())
+        ...     with numpyro.plate("batch", x.shape[0], subsample_size=batch_size):
+        ...         batch_x = numpyro.subsample(x, event_dim=0)
+        ...         batch_y = numpyro.subsample(y, event_dim=0) if y is not None else None
+        ...         mean, rho = net(batch_x)
+        ...         sigma = nn.softplus(rho)
+        ...         numpyro.sample("obs", dist.Normal(mean, sigma), obs=batch_y)
+        >>>
+        >>> n_train_data = 5000
+        >>> x_train, y_train = generate_data(n_train_data)
+        >>> guide = autoguide.AutoDiagonalNormal(model, init_strategy=init_to_feasible)
```

Comment:
You probably want to change this to `init_loc_fn`?

Reply:
Thanks, @neerajprad! I just made the fix. I also fixed the doc of `plate` to not excute usage code, which makes `make doctest` fail in my system (not sure why it passed Github Action).